tiatoolbox/utils/transforms.py

```diff
@@ -307,3 +307,74 @@ def pad_bounds(
         raise ValueError("Invalid number of padding elements.")
     signs = np.repeat([-1, 1], ndims)
     return np.add(bounds, padding * signs)
+
+
+def convert_resolution(input_res, input_unit, baseline_mpp=None, baseline_power=None):
```

Comment:
Something doesn't seem right about this, but I need a while to properly go through it. Also, returning conversion to all units seems a bit like doing a lot of unnecessary work if only one conversion is being used.. Surely a conversion from one unit to another would be better.

Reply:
Thanks @John-P , I hope you find what's not right about this sooner :-D

Re returning conversions to all units: first off, most of the units are needed to be calculated during the calculation of one of them so why not keep all of them?
Also, if we allow requesting for a specific output unit, then have added an unnecessary argument to the function whereas we should still write the function like this (calculate all units) and select the desired unit from the dictionary. Otherwise, the function would comprise many `if` statement to do the right conversion!
```
if input_unit is such and requested_unit is such:
    do this calculation
```
This part should be repeated 4x4=16 times (imaging writing tests for it)! At least, I don't know a better way to write it.
Not to mention, the whole thing really need a small amount of computation (it tooks 12.2 **ns** on my laptop). So I guess we don't need to be worry about it. 
However, if you insist on returning only one desired unit, I'm happy to add another line of code the end of the function, extracting the desired unit from the output dictionary.